modules/standard/Path.chpl

```diff
@@ -421,4 +421,69 @@ proc file.realPath(): string throws {
   return result;
   }
 
+  /* Collapse paths such as `foo//bar`, `foo/bar/`, `foo/./bar`, and 
+   `foo/baz/../bar` into `foo/bar`.  Warning: may alter meaning of paths
+   containing symbolic links.  Similar to :proc:`normCase`, on Windows will replace
+   forward slashes.
+
+   :arg name: a potential path to collapse, possibly destroying the meaning of the
+             path if symbolic links were included.
+   :type name: `string`
+
+   :return: the collapsed version of `name`
+   :rtype: `string`
+  */
+
+  proc normPath(name: string): string {
+
+  if name.isEmptyString() {
+      return name;
+  }
+
+  //stores the result string
+  var result = name;
+
+  if result.find("//") then
+    result = result.replace("//", "/", -1);
+
+  if result.find("/./") then
+    result = result.replace("/./", "/", -1);
+    
+  if result.find("/../") {
+
+    var i = result.find("/../");
+    var j = i-1;
+
+    while(result(j) != "/") {
```

Comment:
This code assumes that it has been passed an absolute path (i.e. one starting with "/").  I strongly suspect it will break for paths like this: `foo/../`

Reply:
Sorry, it assumes either that an absolute path was provide or that `/../` is not the second directory in the path

Reply:
You'll also want to be careful of the case when `/../` or `../` starts your path.  I'm not sure what the second one should do (how does Python respond?), but I think `/../` should just transform into `/` (again, how does Python respond?)